x/oracle/keeper/grpc_query.go

```diff
@@ -136,6 +136,31 @@ func (q querier) MissCounter(
 	}, nil
 }
 
+// SlashWindow queries the current slash window progress of the oracle.
+func (q querier) SlashWindow(
+	goCtx context.Context,
+	req *types.QuerySlashWindow,
+) (*types.QuerySlashWindowResponse, error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "empty request")
+	}
+
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	params := q.GetParams(ctx)
+
+	slashWindow := params.SlashWindow
+	votePeriod := params.VotePeriod
+	currentBlock := ctx.BlockHeight()
+	votePeriodsPerSlashWindow := slashWindow / votePeriod
+
+	currentSlashWindow := uint64(currentBlock) / votePeriodsPerSlashWindow
+	blocksIntoSlashWindow := uint64(currentBlock) - (currentSlashWindow * slashWindow)
```

Comment:
Taking suggestions on variable names here

Reply:
out of ideas :/